### PR TITLE
Add scrollToTop on key navigation links

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

React-router [doesn't manage scroll changes on navigation](https://reactrouter.com/web/guides/scroll-restoration). As a consequence, when users click on an item low in the menu, or on the pagination controls, the new content may reveal up hidden in the non-visible area of the page. This is disturbing for end users.

scrollToTop

On the other hand, the page should not scroll to top everywhere, and especially not when using the back button, applying filters, or switching tabs.

## Solution

- [x] Add a global scrollToTop listener in CoreAdminRouter (resists to layout change)
- [x] Update Edit, Show, Pagination and Menu buttons to enable scrollToTop
- [x] Add scrollToTop to useRedirect
- [x] Add documentation
- [x] Document other buttons, and start fixing stuff down the rabbit hole
- [x] ?!!
- [x] Profit 

scrollToTop2

Supersedes #5891

### CODE REVIEW

**Comment by fzaninotto** [comment]
Switching to RFR

File: docs/Buttons.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+
+| Prop         | Required | Type            | Default            | Description                         |
+| ------------ | -------- | --------------- | ------------------ | ----------------------------------- |
+| `fields`     | Required | `string[]`      | -                  | List of fields to offer sort on     |
+| `icon`       | Optional | `React.element` | `<DeleteIcon>`     | iconElement, e.g. `<CommentIcon />` |
+| `label`      | Optional | `string`        | 'ra.action.delete' | label or translation message to use |
+
+## Record Buttons
+
+### `<DeleteButton>`
```
Comment (jdemangeon) [inline docs/Buttons.md:284]: It seems to lack documentation (or link to other doc) for the 3 buttons here

File: docs/Buttons.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+| `alignIcon`  | Optional | `'left' | 'right`              | `'left'` | Icon position relative to the label     |
+| `children`   | Optional | `React.element`                | -        | icon to use                             |
+| `className`  | Optional | `string`                       | -        | path to link to, e.g. '/posts'          |
+| `color`      | Optional | `'default' | 'inherit'| 'primary' | 'secondary'` | `'primary'` | Label and icon color |
+| `disabled`   | Optional | `boolean`                      | `false`   | If `true`, the button will be disabled |
+| `size`       | Optional | `'large' | 'medium' | 'small'` | `'small'` | Button size                            |
+
+Other props are passed down to [the underlying material-ui `<Button>`](https://material-ui.com/api/button/).
+
+### `<RefreshButton>`
```
Comment (jdemangeon) [inline docs/Buttons.md:305]: Same here, the documentation is missing for miscellaneous buttons

File: packages/ra-core/src/core/CoreAdminRouter.tsx [hunk 3]
```diff
@@ -41,6 +41,20 @@ const CoreAdminRouter: FunctionComponent<AdminRouterProps> = props => {
     const { authenticated } = useAuthState();
     const oneSecondHasPassed = useTimeout(1000);
     const [computedChildren, setComputedChildren] = useSafeSetState<State>([]);
+    const history = useHistory<{ _scrollToTop?: boolean }>();
+    useEffect(
```
Comment (jdemangeon) [inline packages/ra-core/src/core/CoreAdminRouter.tsx:45]: I would extract this part into another hook to split logic

File: docs/Buttons.md (hunk 1 above)
Comment (fzaninotto) [reply docs/Buttons.md:284]: Yes, I chose to not document them now to be able to release the feature not too late. 

I'll work on it "sometime in the future" but not in this PR

File: docs/Buttons.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
+---
+layout: default
+title: "Buttons"
+---
+
+# Buttons
+
+React provides button components for all the common uses.
```
Comment (djhi) [inline docs/Buttons.md:8]:
```suggestion
React-Admin provides button components for all the common uses. 
```

File: docs/Buttons.md [hunk 5]
```diff
@@ -0,0 +1,308 @@
...
+
+**Tip**: If you want to link to the List view manually, use the `/{resource}` location.
+
+## List Buttons
+
+The following buttons are designed to be used in List views.
+
+### `<ExportButton>`
+
+Exports the current list, with filter applied, but without pagination. It relies on [the `exporter` function](./List.md#exporter) passed to the `<List>` component, via the `ListContext`. It's disabled for empty lists.
```
Comment (djhi) [inline docs/Buttons.md:155]:
```suggestion
Exports the current list, with filters applied, but without pagination. It relies on [the `exporter` function](./List.md#exporter) passed to the `<List>` component, via the `ListContext`. It's disabled for empty lists.
```

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by djhi — APPROVED** [review]

**Pull request merged**